File: solving/06-동적계획법(DP)/정수삼각형-DP/11-jiwoung.java
class Solution {
     public static int solution(int[][] triangle) {
        int answer = 0;
        int[][] DP=new int[500][500];
        DP[0][0]=triangle[0][0];
        DP[1][0]=triangle[0][0]+triangle[1][0];
        DP[1][1]=triangle[0][0]+triangle[1][1];

        for (int i=2;i<triangle.length;i++){
            for (int j=0;j<triangle[i].length;j++){
                if (j==0){
                    DP[i][j]=DP[i-1][j]+triangle[i][j];
                }
                else if(j==triangle[i].length-1){
                    DP[i][j]=DP[i-1][j-1]+triangle[i][j];
                }
                else{
                    DP[i][j]=triangle[i][j]+Math.max(DP[i-1][j-1],DP[i-1][j]);
                }

            }

        }

      for (int i=0;i<triangle[triangle.length-1].length;i++){
          if (DP[triangle.length-1][i]>answer){
              answer=DP[triangle.length-1][i];
          }
      }

        return answer;
    }

}

File: solving/N으로표현-DP/11-jiwoung.kt
    var k = 0
    var number: Int = 0
    var resultt: Int = 0

    fun fibo(n: Int, depth: Int): Int {
        if (depth >= 8) {
            resultt = (-1)
            return -1
        } else if (n == number) {
            resultt = n
            Log.d("완성",depth.toString())
            return depth
        }

         else {

            fibo(n + k, depth + 1)
            fibo(n - k, depth + 1)
            fibo(n / k, depth + 1)
            fibo(n * k, depth + 1)
            fibo((n*10)+k, depth + 1)
        }
        Log.d("완성123",resultt.toString())
        return depth

    }


    fun solution(N: Int, number: Int): Int {
        var answer = 0
        k=N
        this.number=number
fibo(N,1)

        return answer

    }
